### ISSUE TITLE
Make ML path selection better: improve accuracy, speed up the inference and reduce the size of jar

### ISSUE THREAD

**Issue by amandelpie** [UnitTestBot/UTBotJava#703]
**Description**

The existing ML path selection is implemented in the utbot-analytics module
It suffers from a few problem:

1. It uses external ML libraries for the model inference. It brings large size of jar
2. It uses Smile library for inference (better to use scikit-learn and provide the model importer)
3. The DJL inference is too slow
4. We measure the metrics on the contest data
5. The utbot-analytics module de-facto is not used

**Expected behavior**

1. Deployed ML models should be a part of jar
2. No external libraries in the utbot-analytics module
3. utbot-analytics module contains only interfaces and pure Kotlin implementations
4. utbot contains separate modules for model inference for the custom inference implementations (like DJL)
5. The new metrics of path selection are created
6. We reached better (significantly) numbers in metrics
7. Obtained models are ranged and well described
8. Training process and hyperparameter tuning is well described and published
